sentry-core/src/main/java/io/sentry/core/ShutdownHookIntegration.java

```diff
@@ -0,0 +1,21 @@
+package io.sentry.core;
+
+import io.sentry.core.util.Objects;
+import org.jetbrains.annotations.NotNull;
+
+/** Registers hook that closes {@link Hub} when main thread shuts down. */
+public final class ShutdownHookIntegration implements Integration {
+
+  private final @NotNull Runtime runtime;
+
+  public ShutdownHookIntegration(final @NotNull Runtime runtime) {
+    this.runtime = Objects.requireNonNull(runtime, "Runtime is required");
+  }
+
+  @Override
+  public void register(@NotNull IHub hub, @NotNull SentryOptions options) {
```

Comment:
```suggestion
  public void register(final @NotNull IHub hub, final @NotNull SentryOptions options) {
```

Reply:
any reason to not apply the `final` modifier?

Reply:
No specific reason, I've seen some methods with `final` and some without. I'll add them.

My personal take on `final` on method parameters is that they clutter code and instead all parameters should be treated as `final` implicitly. That's just an opinion, I will adjust to existing coding standards.

Reply:
got it, just asking cus it was in my first GH "suggestion".

Agreed that they should be treated as `final` implicitly, but IMO when you have multithread systems, having an extra `final` makes the compiler to help you out if you are not cloning/deep copying an object but instead messing up with the references, we had a bug like this a few months ago, so we've decided to add in all the missing ones.

Reply:
Perhaps there is a way to configure static code analysis to make sure it doesn't sneak in?

Reply:
all the data classes are mutable by design, I've tried once but then it warned about almost everything.
so basically what I am doing is, new things I always try to make everything `final` if it can and if I am touching some classes, I already check the missing `final` and `@NotNull`, `@Nullable` to make better interoperability with Kotlin tests